src/components/PaymentDetails.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import moment from 'moment';
+import { Text, View } from 'react-native';
+import { Card } from 'react-native-elements';
+import { onReservationPaymentMethodRead } from '../actions';
+import { Spinner } from './common';
+
+class PaymentDetails extends Component {
+  constructor(props) {
+    super(props);
+
+    const reservation = props.navigation.getParam('reservation', null);
+
+    this.state = { reservation, client: reservation.client || {} };
```

Comment:
si el cliente ya esta en reservation, para que duplicarlo?

Reply:
Esto lo hice porque del lado del cliente y del comercio llega de distinto formato la reservation. Entonces abajo si te fijas tengo `this.state.client.lastName || this.props.lastName` para verificarlo para cada caso. Pero si ponele entra el caso en el que el client no tiene su objeto en la reservations te va a a tirar un error por querer acceder a una propiedad de `undefined`, entonces ese error no te va a tirar porque yo seteo al objeto cliente como un objeto vacío por lo que te va a tirar un false en vez de `undefined`

Reply:
sisi se que viene asi en distinto formato, pero me parecia al pedo igual, directamente podias validar reservation.client antes de intentar acceder a firstname y lastname